Modules/_datetimemodule.c

```diff
@@ -6608,24 +6631,29 @@ _datetime_exec(PyObject *module)
     PyDateTime_Epoch = new_datetime(1970, 1, 1, 0, 0, 0, 0,
                                     PyDateTime_TimeZone_UTC, 0);
     if (PyDateTime_Epoch == NULL) {
+        PyMem_Free(capi);
         return -1;
     }
 
     /* module initialization */
     if (PyModule_AddIntMacro(module, MINYEAR) < 0) {
+        PyMem_Free(capi);
         return -1;
     }
     if (PyModule_AddIntMacro(module, MAXYEAR) < 0) {
+        PyMem_Free(capi);
         return -1;
     }
 
-    x = PyCapsule_New(&CAPI, PyDateTime_CAPSULE_NAME, NULL);
+    x = PyCapsule_New(capi, PyDateTime_CAPSULE_NAME, datetime_destructor);
     if (x == NULL) {
+        PyMem_Free(capi);
         return -1;
     }
 
     if (PyModule_AddObject(module, "datetime_CAPI", x) < 0) {
         Py_DECREF(x);
+        PyMem_Free(capi);
```

Comment:
> At this point, `capi` is owned by the capsule (`x`), so it should be freed only by DECREFing the capsule (which will call `datetime_destructor`) rather than directly by `PyMem_Free`.

Oh, you are right. Thanks.